Userland/Libraries/LibWebView/SingletonProcessState.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteString.h>
+#include <LibCore/Process.h>
+#include <LibCore/StandardPaths.h>
+#include <LibIPC/ConnectionToServer.h>
+#include <LibWebView/SingletonProcessState.h>
+
+namespace WebView {
+
+static HashMap<int, RefPtr<UIProcessConnectionFromClient>> s_connections;
+
+class UIProcessClient final
+    : public IPC::ConnectionToServer<UIProcessClientEndpoint, UIProcessServerEndpoint> {
+    C_OBJECT(UIProcessClient);
+
+private:
+    UIProcessClient(NonnullOwnPtr<Core::LocalSocket>);
+};
+
+ErrorOr<SingletonProcessState::ProcessDisposition> SingletonProcessState::connect(Vector<StringView> const& raw_urls, bool new_window)
+{
+    static constexpr auto process_name = "Ladybird"sv;
+
+    auto runtime_directory = TRY(Core::StandardPaths::runtime_directory());
+    auto socket_path = ByteString::formatted("{}/{}.socket", runtime_directory, process_name);
+    auto pid_path = ByteString::formatted("{}/{}.pid", runtime_directory, process_name);
```

Comment:
We could make the `paths_for_process` thing in Core::Process public as well. Just so we aren't duplicating forming where these paths live